crates/fluvio-benchmark/src/bin/fbm.rs

```diff
@@ -0,0 +1,316 @@
+use std::{
+    fs::File,
+    io::{Read, Write},
+    path::PathBuf,
+    sync::Arc,
+    mem,
+};
+use clap::{arg, Parser};
+use fluvio_cli_common::install::fluvio_base_dir;
+use fluvio_future::{task::run_block_on, sync::Mutex, future::timeout};
+use futures_util::FutureExt;
+use fluvio::Compression;
+use fluvio_benchmark::{
+    benchmark_config::{
+        benchmark_matrix::{
+            BenchmarkMatrix, RecordKeyAllocationStrategy, get_config_from_file, SharedConfig,
+            FluvioProducerConfig, FluvioConsumerConfig, FluvioTopicConfig, BenchmarkLoadConfig,
+        },
+        Seconds, Millis,
+    },
+    benchmark_driver::BenchmarkDriver,
+    stats::{AllStats, AllStatsSync},
+    BenchmarkError,
+};
+
+fn main() {
+    fluvio_future::subscriber::init_logger();
+    let args = Args::parse();
+
+    if args.example_config {
+        print_example_config();
+        return;
+    }
+
+    // TODO accept directory of files.
+    let matrices = if args.test_cluster {
+        test_configs()
+    } else {
+        match args.config {
+            Some(path) => get_config_from_file(&path),
+            None => default_configs(),
+        }
+    };
+
+    let all_stats = Arc::new(Mutex::new(AllStats::default()));
+    let previous = load_previous_stats();
+
+    println!("# Fluvio Benchmark Results");
+    for matrix in matrices {
+        println!("## Matrix: {}", matrix.shared_config.matrix_name);
+        for (i, config) in matrix.into_iter().enumerate() {
+            run_block_on(timeout(
+                config.worker_timeout,
+                BenchmarkDriver::run_benchmark(config.clone(), all_stats.clone()),
+            ))
+            .unwrap()
```

Comment:
is this needed here?

Reply:
The process should quit with failure if the `run_benchmark` times out (the first unwrap) or returns failure (the second unwrap). I could add something custom or have `main()` return a `Result` instead if you prefer.

Reply:
Yes, that should be cleaner